source/Ship.cpp

```diff
@@ -3585,6 +3617,111 @@ void Ship::SetTargetSystem(const System *system)
 
 
 
+// Persistent targets for mission NPCs.
+const bool Ship::HasTravelDirective() const
+{
+	return !stopovers.empty() || destinationPlanet || destinationSystem;
+}
+
+
+
+const map<const Planet *, bool> &Ship::GetStopovers() const
+{
+	return stopovers;
+}
+
+
+
+const bool Ship::AllStopoversVisited() const
+{
+	if(stopovers.empty())
+		return true;
+
+	for(const auto &it : stopovers)
+		if(!it.second)
+			return false;
+	return true;
+}
+
+
+
+const Planet *Ship::GetDestinationPlanet() const
+{
+	return destinationPlanet;
+}
+
+
+
+const System *Ship::GetDestinationSystem() const
+{
+	return destinationSystem;
+}
+
+
+
+// Persistent targets for mission NPCs.
+void Ship::SetDestination(const Planet *destination)
+{
+	destinationPlanet = destination;
+}
+
+
+
+void Ship::SetStopovers(const vector<const Planet *> &stopovers)
+{
+	// Mark each planet as not visited.
+	if(!stopovers.empty())
+		for(const auto &it : stopovers)
+			this->stopovers[it] = false;
```

Comment:
If it's empty, the range based for loop won't do anything anyway.
```suggestion
	// Mark each planet as not visited.
	for(const auto &it : stopovers)
		this->stopovers[it] = false;
```